client/state/nps-survey/actions.js

```diff
@@ -34,21 +35,21 @@ export function setNpsSurveyEligibility( isEligible ) {
 	};
 }
 
+// Undocumented.prototype.checkNPSSurveyEligibility = function( fn ) {
+// 	return this.wpcom.req.get( { path: '/nps' }, { apiVersion: '1.2' }, {}, fn );
+// };
 export function setupNpsSurveyEligibility() {
+	// how is this different from return( dispatch ) => { ??
 	return dispatch => {
```

Comment:
We no longer need a thunk here (or delayed execution), it was previously needed because we were waiting for the http response, in order to dispatch other actions.

This can be simplified to:
```jsx
function setupNpsSurveyEligibility() {
    return  {
        type: NPS_SURVEY_CHECK_ELIGIBILITY
    };
}
```

Reply:
On the component side, you may see the following, where instead of exporting the base component we return a decorated connected component.

The second argument in connect, `mapDispatchToProps` is where dispatch comes back into play. In this example connect will return a connected component that has in it's props a function named `setupNpsSurveyEligibility`. This function instead of returning an action, will dispatch that action immediately when invoked instead.

Internally we're basically calling this when creating the prop:
https://github.com/reduxjs/redux/blob/91aafd8c3fb14f5a1ab689c1fb30270483bed6a3/src/bindActionCreators.js#L1-L5

> If an object is passed, each function inside it is assumed to be a Redux action creator. An object with the same function names, but with every action creator wrapped into a dispatch call so they may be invoked directly, will be merged into the component’s props.

```jsx
//...
//inside of the base component
componentDidMount() {
    this.props.setupNpsSurveyEligibility();
}
//...
export default connect(
	null,
	{ setupNpsSurveyEligibility }
)( MyBaseQueryComponent );
```

https://github.com/reduxjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options